### ISSUE TITLE
3つの推論CLI の CUDA 時間計測方法を統一する

### ISSUE THREAD

**Issue by kurorosu** [kurorosu/pochitrain#200]
## 概要

`pochi infer` / `infer-onnx` / `infer-trt` の3つの推論CLIで CUDA Event による時間計測方法が不統一であり, フェアなベンチマーク比較ができない.

**実測値 (38枚, 512x512, current パイプライン):**

| | pochi infer | infer-trt current |
|---|---|---|
| 純粋推論 | 2.55 ms/image | 0.29 ms/image |
| 全処理 | 10.29 ms/image | 2.88 ms/image |
| **全処理 - 推論** | **7.74 ms** | **2.59 ms** |

推論エンジンの差 (PyTorch vs TensorRT INT8) は当然だが, 推論以外の処理時間の差 (7.74 vs 2.59) はフェアでない. 計測方法の違いがオーバーヘッドとして E2E 時間に乗っている.

### 現状の3者比較

| | pochi infer | infer-onnx | infer-trt |
|---|---|---|---|
| Event作成 | **毎バッチ** (ループ内) | ループ外で1回 | ループ外で1回 |
| 同期方法 | `torch.cuda.synchronize()` | `torch.cuda.synchronize()` | `stream.synchronize()` |
| ウォームアップ | **最初の1バッチのみ** | 10回 + 最初の1バッチ除外 | 10回 + 最初の1枚除外 |

### 問題1: pochi infer が CUDA Event を毎バッチ生成している

```python
# pochitrain/pochi_predictor.py:148-149
# ループ内で毎回 Event を新規作成 → CUDAドライバ呼び出しオーバーヘッド
start_event = torch.cuda.Event(enable_timing=True)
end_event = torch.cuda.Event(enable_timing=True)
```

ONNX/TRT はループ外で1回だけ作成し, 使い回している.

### 問題2: infer-trt が stream.synchronize() を使用している

```python
# infer_trt.py:345
inference.stream.synchronize()  # 特定ストリームのみ (ONNX/pochi inferと異なる)
```

特定ストリーム同期は理論上軽量だが, 他の2つと計測条件が異なるためベンチマーク比較がフェアでない.

### 問題3: pochi infer のウォームアップが不十分

```python
# pochitrain/pochi_predictor.py:138-144
# 最初の1バッチを計測対象外にするのみ (事前ウォームアップなし)
if is_first_batch:
    output = self.model(data)
    if use_cuda:
        torch.cuda.synchronize()
    warmup_samples = batch_size
    is_first_batch = False
```

ONNX/TRT はデータループの前に同一画像で10回のウォームアップを実行している:

```python
# infer_onnx.py:271-285, infer_trt.py:278-292
for _ in range(10):
    inference.run(warmup_np)
```

PyTorch は CUDAカーネルの初回起動コスト (cuDNN auto-tuner, カーネルキャッシュ構築等) が特に高いため, 1回のウォームアップでは不十分. 2バッチ目以降の計測にも暖機コストが残る可能性がある.

## 修正方針

**全CLIを ONNX 方式 (Event ループ外作成 + `torch.cuda.synchronize()` + 10回ウォームアップ) に統一する.**

### 統一後の計測パターン

```python
# ループ外で1回だけ作成
start_event = torch.cuda.Event(enable_timing=True)
end_event = torch.cuda.Event(enable_timing=True)

# ループ内
start_event.record()
推論実行
end_event.record()
torch.cuda.synchronize()
inference_time_ms += start_event.elapsed_time(end_event)
```

### 変更対象

| ファイル | 変更内容 |
|---------|---------|
| `pochitrain/pochi_predictor.py` | Event作成をループ外に移動, 10回ウォームアップ追加 |
| `pochitrain/cli/infer_trt.py` | `event.record(stream)` → `event.record()`, `stream.synchronize()` → `torch.cuda.synchronize()` |
| `pochitrain/cli/infer_onnx.py` | 変更なし (既にこの方式) |

### pochi_predictor.py の修正

```python
# Before (現状)
with torch.no_grad():
    for data, _ in data_loader:
        ...
        if is_first_batch:
            output = self.model(data)
            ...
        else:
            if use_cuda:
                start_event = torch.cuda.Event(enable_timing=True)
                end_event = torch.cuda.Event(enable_timing=True)
                start_event.record()
                output = self.model(data)
                end_event.record()
                torch.cuda.synchronize()
                inference_time_ms += start_event.elapsed_time(end_event)

# After (修正後)
if use_cuda:
    start_event = torch.cuda.Event(enable_timing=True)
    end_event = torch.cuda.Event(enable_timing=True)

with torch.no_grad():
    for data, _ in data_loader:
        ...
        if is_first_batch:
            # ウォームアップ: 最初の1バッチで10回推論
            for _ in range(10):
                self.model(data)
            if use_cuda:
                torch.cuda.synchronize()
            ...
        else:
            if use_cuda:
                start_event.record()
                output = self.model(data)
                end_event.record()
                torch.cuda.synchronize()
                inference_time_ms += start_event.elapsed_time(end_event)
```

### infer_trt.py の修正

```python
# Before (現状)
start_event.record(inference.stream)
inference.execute()
end_event.record(inference.stream)
inference.stream.synchronize()
total_inference_time_ms += start_event.elapsed_time(end_event)

# After (修正後)
start_event.record()
inference.execute()
end_event.record()
torch.cuda.synchronize()
total_inference_time_ms += start_event.elapsed_time(end_event)
```

### 統一後の3者比較

| | pochi infer | infer-onnx | infer-trt |
|---|---|---|---|
| Event作成 | ループ外で1回 | ループ外で1回 | ループ外で1回 |
| 同期方法 | `torch.cuda.synchronize()` | `torch.cuda.synchronize()` | `torch.cuda.synchronize()` |
| ウォームアップ | 10回 + 最初の1バッチ除外 | 10回 + 最初の1バッチ除外 | 10回 + 最初の1枚除外 |

### 補足

- `torch.cuda.synchronize()` は全ストリーム同期のため `stream.synchronize()` より理論上重いが, batch_size=1 の推論では差は無視できるレベル
- 計測のフェアネスを優先し, 3者を同一条件に揃える

## Labels

`enhancement`

## Branch

`fix/unify-cuda-timing`
